app/models/odk/answer_parser.rb

```diff
@@ -41,10 +41,9 @@ def populate_answer_value(answer, content, form_item)
       answer
     end
 
-    def option_id_for_option_node_code(option_node_id)
-      Odk::CodeMapper.instance.item_id_for_code(option_node_id)
-    rescue SubmissionError
-      nil
+    # Get the item ID, unless it's nil (normally that would raise an error).
+    def safe_item_id_for_code(code)
+      code.present? ? CodeMapper.instance.item_id_for_code(code) : nil
```

Comment:
New since pairing. We can't entirely remove `option_id_for_option_node_code` because it's doing some error prevention (CodeMapper raises if `code` is nil). Was this a collective oversight, or did you already suggest a different fix for this?